### PR TITLE
[BEAM-115] Make in-process GroupByKey consistent with Beam model

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by kennknowles** [comment]
Rebased onto new runner location.

R: @tgroh 

I'll grab a committer's attention once you have considered this.

**Comment by dhalperi** [comment]
@kennknowles @tgroh is this important? I was told that InProcess is "done"

**Comment by kennknowles** [comment]
This is part of the movement towards the new Beam runner API. The runner itself is done relative to the general state of things with all the runners. This does not pertain to the backported release.

**Comment by kennknowles** [comment]
R: @dhalperi since you asked and the tests are now passing.

Previously, tests failed and that's why this wasn't in weeks ago. But now with a rebase they are fixed. Likely it was bugs in the runner since I proposed this before it was done.

**Comment by tgroh** [comment]
LGTM

File: runners/direct-java/src/test/java/org/apache/beam/runners/direct/InProcessGroupByKeyOnlyEvaluatorFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+/**
+ * Tests for {@link InProcessGroupByKeyOnlyEvaluatorFactory}.
+ */
+@RunWith(JUnit4.class)
+public class InProcessGroupByKeyOnlyEvaluatorFactoryTest {
+  private BundleFactory bundleFactory = InProcessBundleFactory.create();
+
+  @Test
+  public void testInMemoryEvaluator() throws Exception {
```
Comment (tgroh) [inline runners/direct-java/src/test/java/org/apache/beam/runners/direct/InProcessGroupByKeyOnlyEvaluatorFactoryTest.java]: `testEvaluator`

**Comment by dhalperi** [comment]
LGTM

**Pull request merged**